src/service/resources-impl.js

```diff
@@ -1414,11 +1472,24 @@ export class Resource {
     if (!built) {
       return false;
     }
-    this.state_ = ResourceState_.NOT_LAID_OUT;
+
+    if (this.hasBeenMeasured()) {
+      this.state_ = ResourceState_.READY_FOR_LAYOUT;
+    } else {
+      this.state_ = ResourceState_.NOT_LAID_OUT;
+    }
     return true;
   }
 
   /**
+   * Checks if the current resource has been laid out.
+   * @return {boolean}
+   */
+  hasBeenMeasured() {
```

Comment:
Please use this method in the `measure` as well - there's one more condition with `-10000`